From: Baoquan He <bhe@redhat.com>
Subject: mm/vmalloc: remove the unnecessary type check
Date: Mon, 6 Jun 2022 16:39:05 +0800

Patch series "Cleanup patches of vmalloc".

Some cleanup patches found when reading vmalloc code.


This patch (of 5):

In function adjust_va_to_fit_type(), it checks all values of passed in fit
type, including NOTHING_FIT in the else branch.  In fact, it's unnecessary
to check NOTHING_FIT since the check has been done before
adjust_va_to_fit_type() is called in all call sites.  So clean it up.

Link: https://lkml.kernel.org/r/20220606083909.363350-1-bhe@redhat.com
Link: https://lkml.kernel.org/r/20220606083909.363350-2-bhe@redhat.com
Signed-off-by: Baoquan He <bhe@redhat.com>
Cc: Nicholas Piggin <npiggin@gmail.com>
Cc: Uladzislau Rezki (Sony) <urezki@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/vmalloc.c |    2 --
 1 file changed, 2 deletions(-)

--- a/mm/vmalloc.c~mm-vmalloc-remove-the-unnecessary-type-check
+++ a/mm/vmalloc.c
@@ -1418,8 +1418,6 @@ adjust_va_to_fit_type(struct vmap_area *
 		 * Shrink this VA to remaining size.
 		 */
 		va->va_start = nva_start_addr + size;
-	} else {
-		return -1;
 	}
 
 	if (type != FL_FIT_TYPE) {
_
